### DISCUSSION TITLE
YOLOX and SAHI

### DISCUSSION THREAD

**Discussion by Yash-Ark** [obss/sahi#983] [Q&A]
Hello Everyone,

I am working on a project where the input image size is 7296 x 5472 and 2160 x 3840. The image contains minor elements in the large-scale image; hence, we can't resize it.

In light of this, I discovered the concept of the SLIDING WINDOW Algorithm and found SAHI, which provides the same features to achieve the goal I am aiming for.

Since I developed the model on YOLOX-X (with some specific instructions, so I can't use other detection models) and want to leverage the functionality of SAHI, I am looking for the implementation, guide, or manual—anything that can help me develop the end-to-end machine learning pipeline for SAHI + YOLOX.

Here is what I have done so far:

1. Utilized the Slice Image Module of SAHI as a preprocessing step and created multiple patches of the large image.
2. Have YOLOX model and its inference code.

What I am looking for:

1. Is there an already implemented solution available? If so, please share.
2. If not the above, how can I use the detections of patches in SAHI's post-process?
   2.1 Source for any good explanation on SAHI Post-process.

Thank you; I really appreciate your time and efforts.

Please let me know if there is anything else needed from my side.

**Comment by aronnoordhoek** [comment]
This? https://docs.ultralytics.com/guides/sahi-tiled-inference/

**Reply by Yash-Ark** [reply]
Thank you for your response @aronnoordhoek But I am working with YOLOX and the link you gave is referring to YOLOv8. YOLOX and YOLOv8 model inference is quite different. I already have YOLOv5 model inference but while trying for YOLOX the process is not the same. That's why I am seeking for help.

**Reply by aronnoordhoek** [reply]
Ah my bad I thought you were referring to the X variants of the YOLO models. It refers to the sizing.
